From fce1836eadc9a50c9572c84b10a85671c22811c9 Mon Sep 17 00:00:00 2001
From: Li Jun <jun.li@nxp.com>
Date: Sat, 19 Aug 2017 03:11:57 +0800
Subject: [PATCH 2453/5242] MLK-16013-44 staging: typec: tcpm: set data role
 after src detach

commit  04cabfd329b81990756f9653a4f212e9dca586ee from
https://source.codeaurora.org/external/imx/linux-imx.git

As the default data role is device mode if the port is not host,
so set the port data role to be device mode after src detach, this
is to fix the issue of port data role still kept to be host while
the port is open.

Acked-by: Peter Chen <peter.chen@nxp.com>
Signed-off-by: Li Jun <jun.li@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/usb/typec/tcpm.c |    1 +
 1 file changed, 1 insertion(+)

diff --git a/drivers/usb/typec/tcpm.c b/drivers/usb/typec/tcpm.c
index 34d1cab..0fa3038 100644
--- a/drivers/usb/typec/tcpm.c
+++ b/drivers/usb/typec/tcpm.c
@@ -2673,6 +2673,7 @@ static void tcpm_detach(struct tcpm_port *port)
 
 static void tcpm_src_detach(struct tcpm_port *port)
 {
+	port->data_role = TYPEC_DEVICE;
 	tcpm_detach(port);
 }
 
-- 
1.7.9.5

